db-checkbook/build_scripts/01_dataloading.py

```diff
@@ -1,16 +1,33 @@
-from dcpy.connectors.s3 import client
+from dcpy.connectors import s3
+from functools import wraps
+from pathlib import Path
+import pandas as pd
+import geopandas as gpd
 
+base_bucket = 'edm-recipes'
+base_url = "https://edm-recipes.nyc3.cdn.digitaloceanspaces.com"
 
-def do_stuff() -> None:
-    print("doing stuff")
-    s3_client = client()
-    available_buckets = [
-        bucket["Name"] for bucket in s3_client.list_buckets()["Buckets"]
-    ]
-    print(f"This S3 client is a {type(s3_client)}")
-    print(f"This S3 client has access to buckets: {available_buckets}")
+
+
+# reading in the data: using S3 connectors
+
+# example: datasets/dcp_cpdb/2018_adopted_polygons/
+# dataset name: datasets/dcp_cpdb
+# version: 2018_adopted, 2019_adopted, 2020_adopted, 2021_adopted, 2022_adopted, 2023_executive
+# type geom: _polygons, _points
+
+def read_s3_edm_recipes_cpdb(dataset_name, version, type_geom, save_file_path):
+    digital_ocean_filepath = f'{dataset_name}/{version}{type_geom}/dcp_cpdb.shp'
+    s3_client = s3.client()
+    file = s3_client.download_file(base_bucket, digital_ocean_filepath, save_file_path)
+    return file
+
+
+def read_edm_recipes_nyc_checkbook(name):
+    file_name = f'{base_url}/{name}'
```

Comment:
couple questions here - is this going to work as planned? From my understanding, you'd need to feed in `"datasets/nycoc_checkbook/latest/nycoc_checkbook.csv"` as the name, which seems like maybe we should only need to feed in `"nycoc_checkbook"`.

Similarly, for function name - this doesn't seem specific to nycoc checkbook, but rather is reading in any csv of any recipe, so it should be named as such.

And version should be specified as an argument, just like for the cpdb function.

Let me know if I'm making any incorrect assumptions about the data being pulled in

Reply:
For version as an argument, should it be implemented when we have the Checkbook API pulling also set up? At the moment we are working with a single dataset, but we could also just do a default of version = "" for now.

Reply:
Hmm that's a good point, I was assuming the checkbook data would just be one file but it won't necessarily be that way in the future...

Let's go ahead and make that assumption for now, and for this go with what I'm suggesting. I think that's how the checkbook data in recipes will be most useful in the future anyways, even if it requires some aggregating in the data library step (and thinking about not being super wasteful with storage since this is one of the bigger source data sets in terms of rows)

Reply:
So "latest" still being default version